Network/Units/Client/Client.cpp

```diff
@@ -6,3 +6,138 @@
+    _messageId = 0x0000;
+    _token = 0;
+    _io_thread = nullptr;
+}
+
+Client::~Client()
+{
+    _io_context->stop();
+    if (_io_thread->joinable()) {
+        _io_thread->join();
+    }
+}
+
+bool Client::connectToServer(std::string ipServer, int tcp_port)
+{
+    try {
+        _serverIp = ipServer;
+        _tcpPort = tcp_port;
+        asio::ip::tcp::resolver resolver(*_io_context);
+        asio::ip::tcp::resolver::results_type endpoints = resolver.resolve(_serverIp, std::to_string(_tcpPort));
+        asio::connect(*_tcp_socket, endpoints);
+
+        asio::error_code ec;
+        asio::read(*_tcp_socket, asio::buffer(&_udpPort, sizeof(_udpPort)), ec);
+        asio::read(*_tcp_socket, asio::buffer(&_token, sizeof(_token)), ec);
+        // asio::read(*_tcp_socket, asio::buffer(&_idxPlayerComponent, sizeof(_idxPlayerComponent)), ec);
+        std::cout << "Connected to server with token: " << "0x" << std::hex << std::setw(8) << std::setfill('0') << _token << std::dec << std::endl;
+        // std::cout << "Connected to server with index player component: " << _idxPlayerComponent << std::endl;
+
+        _udp_socket->open(asio::ip::udp::v4());
+        _server_endpoint = std::make_shared<asio::ip::udp::endpoint>(asio::ip::address::from_string(_serverIp), _udpPort);
+
+        _startReceive();
+
+        _io_thread = std::make_shared<std::thread>([this]() { _io_context->run(); });
+        _io_thread->detach();
+        return true;
+    } catch (const std::exception& e) {
+        std::cerr << "Error during connection: " << e.what() << std::endl;
+        return false;
+    }
+}
+
+void Client::_addPacketToQueueReceived(UDPPacket packet)
+{
+    std::lock_guard<std::mutex> lock(_mutex);
+    _queueMessage.push_back(packet);
+}
+
+void Client::_startReceive()
+{
+    _udp_socket->async_receive_from(asio::buffer(_recvBuffer), *_server_endpoint,
+        [this](const asio::error_code& error, std::size_t bytes_recvd) {
+            if (!error && bytes_recvd > 0) {
+                try {
+                    std::string message(_recvBuffer.data(), bytes_recvd);
+
+                    UDPPacket packet(message);
+
+                    _addPacketToQueueReceived(packet);
+
+                } catch (const std::exception& e) {
+                    std::cerr << "Exception in message handler: " << e.what() << std::endl;
+                }
+            }
+            asio::post(_udp_socket->get_executor(),
+                [this]() {
+                    _startReceive();
+                });
+        });
+}
+
+void Client::sendMessage(std::vector<uint8_t> message, uint32_t token)
+{
+    (void) token;
+    try {
+        _udp_socket->send_to(asio::buffer(message), *_server_endpoint);
+    } catch (const std::exception& e) {
+        std::cout << "Error in send Message client: " << e.what() << std::endl;
+    }
+}
+
+int Client::getNumberClient()
+{
+    throw ABIError("A client unit can't have client connected.");
+}
```

Comment:
You can maybe declare it as default instead of throwing something since it's not an error, it just can't be possible

Reply:
I can't set this function as delete because it's a child method of INetworkUnit.